### ISSUE TITLE
Cloudformation package support --acl

### ISSUE THREAD

**Issue by dipthegeezer** [aws/aws-cli#2681]
Hi

I have a use case where I am using cloudformation package to upload source code to an S3 bucket. However this bucket is set up for cross account access. This seems to cause an issue when the account the bucket is in tries to access the source code as the bucket owner does have full control of the object. Would it be possible to add the --acl flag which is available for the S3 command and all the options that are allowed with that flag. My specific scenario is that account A is used to build and test the code however it uploads the source to Account B. Account B then deploys the code via codepipeline.

Thanks

**Comment by stealthycoin** [comment]
Marking as a feature request.

**Comment by ASayre** [comment]
Good Morning!

We're closing this issue here on GitHub, as part of our migration to [UserVoice](https://aws.uservoice.com/forums/598381-aws-command-line-interface) for feature requests involving the AWS CLI.

This will let us get the most important features to you, by making it easier to search for and show support for the features you care the most about, without diluting the conversation with bug reports.

As a quick UserVoice primer (if not already familiar): after an idea is posted, people can vote on the ideas, and the product team will be responding directly to the most popular suggestions.

We’ve imported existing feature requests from GitHub - Search for this issue there!

And don't worry, this issue will still exist on GitHub for posterity's sake.  As it’s a text-only import of the original post into UserVoice, we’ll still be keeping in mind the comments and discussion that already exist here on the GitHub issue.

GitHub will remain the channel for reporting bugs.  

Once again, this issue can now be found by searching for the title on: https://aws.uservoice.com/forums/598381-aws-command-line-interface 

-The AWS SDKs & Tools Team

**Comment by jamesls** [comment]
Based on community feedback, we have decided to return feature requests to GitHub issues.

**Comment by barbaraokupna** [comment]
Yes. Same here. It would be really useful.

**Comment by justnance** [comment]
Thanks for the feedback.  @sanathkr - what are your thoughts?

**Comment by michielvd** [comment]
Hi, Any progress/feedback on this? I would also need this. As we separate our build process form deployment resources needed in other accounts.

**Comment by giles-jason** [comment]
Agreed. This is a huge headache for us as well. It is disappointing that this issue has existed for almost 4 years without much movement. 

@justnance to address your earlier question, the issue we run into for our team is the fact that we have our CI/CD infrastructure located in an account dedicated to those activities. Our environments are run out of separate accounts and the default ACL applied to objects that are uploaded to our artifacts buckets causes CloudFormation to be unable to read these resources when orchestrating deployments in other accounts.

**Comment by jewelsjacobs** [comment]
I need this as well. Status?

**Comment by tim-finnigan** [comment]
We have been advised to refer users of `cloudformation package` to the [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html) and [sam-package](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-package.html) command. For this specific use case, it looks like there is already a related feature request in the SAM CLI GitHub repository: https://github.com/aws/aws-sam-cli/issues/2195. There is also a PR for that feature request but it was recently closed due to inactivity

**Comment by tjstansell** [comment]
Does that mean that `cloudformation package` is being deprecated?  That seems like a step backwards.  I'm personally not a fan of the SAM tools.  That said, with S3 buckets now supporting `BucketOwnerEnforced` ownership, ACLs are no longer a problem for us.  Of course, we're also mostly moving to terraform, so we care less about cloudformation than we used to.

**Comment by tim-finnigan** [comment]
Thanks for the feedback - I wouldn't say that the CloudFormation customization is being deprecated but it is de-prioritized as of now because the team assigned to maintain it is currently focusing their efforts on the SAM CLI. I think many users have also switched to the AWS CDK for generating CloudFormation going forward.

**Comment by lorengordon** [comment]
The package functionality in aws cli, and the generate functionality in CDK are entirely different things, aren't they? It's obviously subjective either way, without data, but I wouldn't want to jump to a conclusion about how many looking to use this package functionality have instead switched to CDK as a replacement.

**Comment by tim-finnigan** [comment]
@lorengordon thanks for clarifying - yes that's true those are different use cases, I just wanted to bring up CDK as Terraform was mentioned above and the CDK is the AWS infrastructure-as-code approach for generating CloudFormation at a higher level of abstraction going forward. But the SAM CLI offers the [sam package](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-package.html) command as an alternative.
